src/controllers/roomSocket.ts

```
@@ -82,16 +98,27 @@ function joinRoom(socket: Socket, data: joinRoomData) {
+        // Recalculate spending limits for all gods
+        if (data.joinAs === 'god') {
+            const updatedRoom = await clientDB.collection('rooms').findOne({code: data.code});
+            const newSpendingLimit = Math.floor(updatedRoom.bank / updatedRoom.gods.length);
+            await clientDB.collection('rooms').updateMany(
+                {code: data.code},
+                {$set: {'gods.$[].spendingLimit': newSpendingLimit}}
+            );
+        }
 
         const updatedRoom = await clientDB.collection('rooms').findOne({code: data.code});
         if (updatedRoom && updatedRoom.creator) {
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)